install.sh

```diff
@@ -1174,13 +1174,28 @@ create_server_user_if_needed() {
 
   # OK! Let's proceed.
   #
-  # useradd comes from the passwd package, whereas adduser is a Debian-specific utility
-  # <https://packages.debian.org/adduser>, so we prefer useradd here. Per the man page for useradd,
-  # USERGROUPS_ENAB in /etc/login.defs controls if useradd will automatically create a group for
-  # this user (the new group would have the same name as the new user). On systems such as OpenSuSE
-  # where that flag is set to false by default, or on systems where the administrator has personally
-  # tuned that flag, we need to provide --user-group to useradd so that it creates the group.
-  useradd --system --user-group "$SERVER_USER"
+  # To create the server user, we first try `useradd`, which is widely available on most
+  # distros. If that isn't available, it's likely we're running on a busybox based system.
+  # busybox provides an `adduser` command, so if the busybox binary exists we try that
+  # instead.
+  #
+  # Note that debian provides an `adduser` command as well, but its usage is different.
+  # useradd is available on debian anyway, so we'll end up using that.
+  if which useradd >/dev/null; then
+    # Per the man page for useradd, USERGROUPS_ENAB in /etc/login.defs controls if useradd
+    # will automatically create a group for this user (the new group would have the same
+    # name as the new user). On systems such as OpenSuSE where that flag is set to false
+    # by default, or on systems where the administrator has personally tuned that flag,
+    # we need to provide --user-group to useradd so that it creates the group.
+    useradd --system --user-group "$SERVER_USER"
+  elif which busybox >/dev/null; then
```

Comment:
Hmm, `busybox` is commonly installed on normal Linux systems, so the existence of `busybox` doesn't necessarily mean that we're currently running under it. Is there a better way to detect this? Or maybe you really want to check for the presence of `adduser` and `addgroup`?

Reply:
Well, it should only fall back to this if it can't find `useradd`. I decided to check for busybox rather than `adduser`/`addgroup` since the latter might potentially be other (incompatible) commands with those names. Perhaps we should call `busybox adduser` to make sure we're not running some other command.

I think running the busybox versions when available should be ok, even if it's not the system default -- they should make the same modifications to `/etc/passwd` and friends, no?

Reply:
On my system, `busybox adduser` doesn't work:

```
$ busybox adduser
adduser: applet not found
```

And it looks like your usage here is indeed incompatible with the Debian adduser / addgroup commands, which have no short flags, only long flags.

Maybe you can detect busybox like this? https://stackoverflow.com/a/24884579